src/callable/operators.rs

```diff
@@ -287,7 +287,13 @@ impl Callable for InfixColon {
                 return Error::Other("Cannot increment by 0".to_string()).into();
             }
 
-            if (end - start) / by < 0.0 {
+            let range = end - start;
+
+            if range % by != 0.0 {
+                return Error::Other("Argument 'by' does not divide range.".to_string()).into();
+            }
```

Comment:
Let's hold off on this assertion. In my experience, when specifying a `by` step, it's common that your start or end might be variable - you may care that you start at some number and increment by some amount, but just want to reach an end. For example:

```r
x <- [1, 2, 3, 4]
odd_idxs <- 1:2:length(x)
even_idxs <- 2:2:length(x)
x[odd_idxs]
x[even_idxs]
```

Regardless of the length of x, we only care that we've split based on whether the index is odd or not.